src/lib/operators/jit_operator/operators/jit_write_reference.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "abstract_jittable_sink.hpp"
+
+namespace opossum {
+
+/* JitWriteReference must be the last operator in any chain of jit operators.
+ * It is responsible for
+ * 1) adding column definitions to the output table
+ * 2) appending the current row id to the current output chunk
+ * 3) creating a new output chunk and adding output chunks to the output table
+ */
+class JitWriteReference : public AbstractJittableSink {
+ public:
+  struct OutputColumn {
+    const std::string column_name;
+    const ColumnID referenced_column_id;
+  };
+
+  std::string description() const final;
+
+  std::shared_ptr<Table> create_output_table(const Table& in_table) const final;
```

Comment:
Please add some documentation here. The interactions of this and `add_output_column` are not clear - I would expect a method `create_output_table` to also create the output columns.

Reply:
Add comments. I also added further comments in `JitReadTuples`.